bridge/src/main/java/io/stargate/bridge/impl/BridgeImpl.java

```diff
@@ -71,9 +71,7 @@ public BridgeImpl(
             EXECUTOR_SIZE, GrpcUtil.getThreadFactory("bridge-stargate-executor", true));
     server =
         NettyServerBuilder.forAddress(new InetSocketAddress(listenAddress, port))
-            // `Persistence` operations are done asynchronously so there isn't a need for a separate
-            // thread pool for handling gRPC callbacks in `GrpcService`.
-            .directExecutor()
+            .executor(Executors.newScheduledThreadPool(16))
```

Comment:
I don't know the context here, but just in general 16 sounds like a relatively small thread pool?
Not sure what would be a good way to figure out an optimal setting; would be good to make it configurable I think, with a new `stargate.rpc.xxx` property.

Reply:
We need to understand first why is better performance observed and if we are not mistaken.. Why was a direct executor chosen at the first place..

We also need to have a proper way of getting the default value for the pool size, should it be based on the number of cores or something like this.. Plus, we should have it configurable..

This is very sensitive change for me and I would like to avoid doing any damage long term..

Reply:
> Why was a direct executor chosen at the first place..

Reply:
@mpenick OK, but what about the requests that are not jumping to another thread? For example, `DescribeKeyspace` call is not jumping to another thread right? Can this be the reason? Maybe we need to offload those call also to another thread?

Reply:
Just to track the discussion we had @ivansenic. Ivan and I talked about making another, explicitly created, thread pool for blocking requests like `DescribeKeyspace` and still continuing to use `.directExecutor()`. That way the already async requests like "execute" type request don't have to move through yet another thread.

Reply:
I will implement that here, cool